src/kpro_sent_reqs.erl

```diff
@@ -0,0 +1,118 @@
+%%%
+%%%   Copyright (c) 2014-2018, Klarna AB
+%%%
+%%%   Licensed under the Apache License, Version 2.0 (the "License");
+%%%   you may not use this file except in compliance with the License.
+%%%   You may obtain a copy of the License at
+%%%
+%%%       http://www.apache.org/licenses/LICENSE-2.0
+%%%
+%%%   Unless required by applicable law or agreed to in writing, software
+%%%   distributed under the License is distributed on an "AS IS" BASIS,
+%%%   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%%%   See the License for the specific language governing permissions and
+%%%   limitations under the License.
+%%%
+
+%%%=============================================================================
+%%% @doc
+%%% This module manages an opaque of sent-request collection.
+%%% @end
+%%% ============================================================================
+
+%% @private
+-module(kpro_sent_reqs).
+
+%%%_* Exports ==================================================================
+
+%% API
+-export([ new/0
+        , add/5
+        , del/2
+        , get_req/2
+        , get_corr_id/1
+        , increment_corr_id/1
+        , scan_for_max_age/1
+        ]).
+
+-export_type([requests/0]).
+
+-type corr_id() :: kpro:corr_id().
+-define(REQ(Caller, Ref, API, Vsn, Ts), {Caller, Ref, API, Vsn, Ts}).
+-type req() :: ?REQ(pid(), reference(), kpro:api(), kpro:vsn(), integer()).
+
+-record(requests,
+        { corr_id = 0
+        , sent = #{} :: #{corr_id() => req()}
+        }).
+
+-opaque requests() :: #requests{}.
+
+-include("kpro_private.hrl").
+
+
+%%%_* APIs =====================================================================
+
+-spec new() -> requests().
+new() -> #requests{}.
+
+%% @doc Add a new request to sent collection.
+%% Return the last corrlation ID and the new opaque.
```

Comment:
Nit: s/corrlation/correlation/g
Also "opaque" is not a noun, so end the sentence with sth like "new opaque collection" or just "new collection".